google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -106,15 +106,22 @@ private AppEngineUtil() {
       @NotNull Project project) {
     List<ModuleDeploymentSource> moduleDeploymentSources = Lists.newArrayList();
 
+    boolean hasNonAppEngineStandardJavaModules = false;
+
     for (Module module : ModuleManager.getInstance(project).getModules()) {
-      if (!containsAppEngineStandardArtifacts(project, module)
-          && isJarOrWarMavenBuild(project, module)) {
-        moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module));
+      if (ModuleType.is(module, JavaModuleType.getModuleType())) {
+        if (!containsAppEngineStandardArtifacts(project, module)) {
+          hasNonAppEngineStandardJavaModules = true;
+          if (isJarOrWarMavenBuild(project, module)) {
+            moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module));
+          }
+        }
       }
     }
 
-    Artifact[] artifacts = ArtifactManager.getInstance(project).getArtifacts();
-    if (!containsAppEngineStandardArtifacts(project, Arrays.asList(artifacts))) {
+    // I kind of expected the logic for adding the user specific deployment source to be in the
+    // artifact based deployment sources.
```

Comment:
With regard to the comment: I _think_ it could be an artifact deployment source (currently `UserSpecifiedDeploymentSource` is a module deployment source which is why its here) . But thinking about this more, it might make sense to even create a new type of DeploymentSource (neither module, nor artifact). We can look at this in a separate PR to see if that makes sense.